src/app/content/components/Page/contentDOMTransformations.ts

```diff
@@ -42,6 +46,40 @@ function removeDocumentTitle(rootEl: HTMLElement) {
   ].join(',')).forEach((el) => el.remove());
 }
 
+// set the top heading's level to topHeadingLevel and adjust other headings accordingly
+function changeHeadingLevels(document: Document, rootEl: HTMLElement, topHeadingLevel: number) {
+  const headingLevels = [ 1, 2, 3, 4, 5, 6 ];
+  const currentTopHeading = headingLevels.find((level) => rootEl.querySelectorAll(`h${level}`)?.length);
+
+  if (!currentTopHeading || topHeadingLevel <= currentTopHeading) {
```

Comment:
if desired topHeading level is 2, and the currentTopHeading is 3, we don't want to skip we want to bump them all up.

if they are equal we can return

Reply:
ah ok, was confused on this point